### PR TITLE
Add features for skipping parsing

### PR DESCRIPTION
**Pull request by cossssmin**
## Proposed Changes

This PR adds two new features that allow outputting content as-is, basically skipping any parsing normally done by the plugin.

This is useful when you need to use the same syntax with other templating libraries.

### `<raw>` tag

You can now use the `<raw>` tag (configurable) to tell `posthtml-expressions` to skip parsing of everything inside it.

```html
<raw>
  <if condition="foo === 'bar'">
    <p>Output {{ foo }} as-is</p>
  </if>
</raw>
```

```html
<if condition="foo === 'bar'">
  <p>Output {{ foo }} as-is</p>
</if>
```

You can customize the name of the tag:

```js
var opts = {
  ignoredTag: 'verbatim', 
  locals: { foo: 'bar' } }
}

posthtml(expressions(opts))
  .process(readFileSync('index.html', 'utf8'))
  .then((result) => console.log(result.html))
```

```html
<verbatim>
  <if condition="foo === 'bar'">
    <p>Output {{ foo }} as-is</p>
  </if>
</verbatim>
```

```html
<if condition="foo === 'bar'">
  <p>Output {{ foo }} as-is</p>
</if>
```

### @ symbol

For one-offs where you need to output delimiters as-is and prevent the plugin from parsing expressions inside them, you can now use the [Blade-inspired](https://laravel.com/docs/5.8/blade#blade-and-javascript-frameworks) `@` symbol in front of them:

```html
<p>Here, @{{ foo }} will be output as it is.</p>
```

```html
<p>Here, {{ foo }} will be output as it is.</p>
```

## Types of Changes

- [ ] Bug (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature which changes existing functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) guide
- [x] Lint and unit tests pass with my changes
- [x] I have added tests that prove my fix is effective/works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes are merged and published in downstream modules

### CODE REVIEW

**Comment by cossssmin** [comment]
I think it's useful to have them both, as they cover different cases.

You use `<raw>` to wrap it around other tags, like entire HTML blocks.

The `@` symbol is only for ignoring expressions. For example, it's better suited in cases like `<p class="@{{ dynamicClass }}">`, where you wouldn't be able to use `<raw>`.

**Comment by anikethsaha** [comment]
`<p class="@{{ dynamicClass }}">`

will output 

`<p class="{{ dynamicClass }}">` right ?

I think we can do this using 

`<raw><p class="{{ dynamicClass }}"></raw>` right ?

**Comment by cossssmin** [comment]
You may have something in there that you _still_ need to parse:

```html
<raw><p class="@{{ dynamicClass }}">{{ iWantThisVariableParsed }}</p></raw>
```

Blade approaches this in the [same way](https://laravel.com/docs/5.8/blade#blade-and-javascript-frameworks), it uses `@{{ }}` and the there's a `@verbatim ... @endverbatim` directive (equivalent to our `<raw>` tag).

**Comment by anikethsaha** [comment]
```html
<raw><p class="@{{ dynamicClass }}">{{ iWantThisVariableParsed }}</p></raw>
```

What will be the output for this ?

I think it will be 

```html
<p class="@{{ dynamicClass }}">{{ iWantThisVariableParsed }}</p>
```

and in the next iteration, it will resolve the `@{{` also, right?

```html
<p class="{{ dynamicClass }}">{{ iWantThisVariableParsed }}</p>
```

can you add this in test as well ! or similar if not present !

**Comment by cossssmin** [comment]
Updated the `raw` test.

> and in the next iteration, it will resolve the `@{{` also, right?
> 
> ```
> <p class="{{ dynamicClass }}">{{ iWantThisVariableParsed }}</p>
> ```

No, it should not (and it doesn't), because everything inside `<raw>` should be discarded (ignore and move on to the next node)

**Comment by anikethsaha** [comment]
ok but still I am not getting a use case where it cant be solved using `<raw>` . may be I am missing something. can you provide any example !

**Comment by cossssmin** [comment]
I just did above, not sure how to make it clearer, but let me try again...

Basically, there are situations where you can't use a `<raw>` tag, like inside an attribute.

Imagine passing this `locals` object:

```js
locals: {
  foo: 'bar',
}
```

With `@` symbol:

```html
<p class="@{{ dynamicClass }}">{{ foo }}</p>
```

```html
<p class="{{ dynamicClass }}">bar</p>
```

If we wrap with `<raw>`, then `foo` is no longer evaluated, although we want it to:

```html
<raw><p class="@{{ dynamicClass }}">{{ foo }}</p></raw>
```

Result:

```html
<p class="@{{ dynamicClass }}">{{ foo }}</p>
```

Makes sense now?

**Comment by anikethsaha** [comment]
ok got it !

Well it make sense 👍

**Comment by Scrum** [comment]
@cossssmin @anikethsaha Thank you for the work you have done - it is very valuable for our team.

Publish v1.2.0
